### PR TITLE
Ptx 23988

### PR DESCRIPTION
**Pull request by vprabhakar-px**
Functions to support Skinny Snaps and ignore Add disk test when DMThin is used

### CODE REVIEW

**Comment by vprabhakar-px** [comment]
https://aetos.pwx.purestorage.com/resultSet/testSetID/640954 :  Test run after skipping all Add disk tests on DMThin

**Comment by vprabhakar-px** [comment]
Script run output 

2024-05-11 22:29:14 +0530:[INFO] [{DeletePoolAndValidateMultipath}] [basic.glob..func281.2:#12494] - Multipath Devices [[{3624a93701b07343bc7f64e4400014c2d dm-1 350G active PURE [{sdg active ready running} {sdh active ready running} {sdi active ready running} {sdj active ready running}]} {3624a93701b07343bc7f64e4400014c2c dm-0 250G active PURE [{sdc active ready running} {sdf active ready running} {sdd active ready running} {sde active ready running}]}]]

Cloud Drives On Node [px2-release-facd-3-1-2-u20-k540-146-1] [&{map[6ea7951aca:{pure-block 250 6ea7951aca 6294aaba-e4aa-4ead-be37-f656e4d2f372 /dev/mapper/3624a93701b07343bc7f64e4400014c2c 0 0 data In Use map[portworx.io/flasharray-id:1b07343b-c7f6-4e44-849d-8b5e8cdeb16c portworx.io/flasharray-management-endpoint:10.13.57.32 portworx.io/flasharray-name:mh531-f11] <nil>  } a3a4824ca9:{pure-block 350 a3a4824ca9 6f030414-dd17-4cec-ac4f-97451d61e478 /dev/mapper/3624a93701b07343bc7f64e4400014c2d 0 0 data In Use map[portworx.io/flasharray-id:1b07343b-c7f6-4e44-849d-8b5e8cdeb16c portworx.io/flasharray-management-endpoint:10.13.57.32 portworx.io/flasharray-name:mh531-f11] <nil>  }] aa88d981-ef45-496e-a534-4f24e62c5354  px2-release-facd-3-1-2-u20-k540-146-1 2 2024-05-11 01:40:05.217900038 +0000 UTC px2-release-facd-3-1-2-u20-k540-146-1 default In Use map[]}]

**Comment by vprabhakar-px** [comment]
latest runs : https://jenkins.pwx.dev.purestorage.com/job/DMTHIN/job/DMThin-OCP-FACD/job/dmthin-ocp-facd-systemtest/27/console

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -141,33 +144,31 @@ var _ = Describe("{StoragePoolExpandDiskAdd}", func() {
...
-				if !isPoolAddDiskSupported {
-					IsExpectederr := strings.Contains(err.Error(), "add-drive type expansion is not supported with px-storev2. Use resize-drive expansion type")
-					dash.VerifyFatal(IsExpectederr, true, err.Error())
-					log.InfoD("Drive add not supported :%s, hence skipping the test", err.Error())
-					Skip("drive add to existing pool not supported for px-storev2 or px-cache pools")
-
+			if IsPoolAddDiskSupported() {
+				err = Inst().V.ExpandPool(poolIDToResize, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				if err != nil {
+					log.FailOnError(err, "is Pool Expand using Add disk successful ?")
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:150]: both log.FailonError and dash.VerifyFatal are not required. one is enough

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -3760,28 +3761,20 @@ var _ = Describe("{PoolMaintenanceModeAddDisk}", func() {
...
-					log.InfoD("Drive add not supported :%s, hence skipping the test", err.Error())
-					Skip("drive add to existing pool not supported for px-storev2 or px-cache pools")
-
-				}
+			if IsPoolAddDiskSupported() {
+				log.InfoD("Current Size of the pool %s is %d", poolToBeResized.Uuid, poolToBeResized.TotalSize/units.GiB)
+				err = Inst().V.ExpandPool(poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, true)
+				log.FailOnError(err, "Pool Expand using Add Disk Failed ")
 
+				dash.VerifyFatal(err, nil, "Pool expansion init successful?")
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:3769]: same here

File: tests/basic/storage_pool_test.go (hunk 1 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:150]: my bad !! fixed

File: tests/basic/storage_pool_test.go (hunk 2 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:3769]: fixed

**Comment by vprabhakar-px** [comment]
https://aetos.pwx.purestorage.com/resultSet/testSetID/641462

File: tests/common.go [hunk 3]
```diff
@@ -12536,3 +12554,235 @@ func GetAllMultipathDevicesPresent(n *node.Node) ([]MultipathDevices, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	for _, eachMultipathDev := range multiPathDevices {
+		multiPathDev = append(multiPathDev, eachMultipathDev.DevId)
+	}
+	return multiPathDev, nil
+}
+
+// CreateNameSpace Creates new NameSpace
```
Comment (sayalasomayajula-px) [inline tests/common.go:12571]: there is already a function with same name in common.go , can we change the name of the function..

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm - Pls take care of the minor comment added before merging

**Review by sayalasomayajula-px — APPROVED** [review]

**Pull request merged**